script.reddit.reader/default.py

```diff
@@ -0,0 +1,2201 @@
+    #    func = funcs[selected_index]
+    #     #assign item from funcs to func
+    #    func(subreddit, name, type)    
+    log('selected_index ' + str(selected_index))
+    if selected_index == 0:       # 0->first item
+        addSubreddit('','','')
+        pass
+    elif selected_index == 1:     # 1->second item
+        editSubreddit(subreddit,'','')
+        pass
+    elif selected_index == 2:     # 2-> third item
+        removeSubreddit(subreddit,'','')
+        pass
+    else:                         #-1 -> escape pressed or [cancel]
+        pass
+    
+    xbmc.executebuiltin( "Dialog.Close(busydialog)" )
+    #this is a hack to force the subreddit listing to refresh.
+    #   the calling gui needs to close after calling this function 
+    #   after we are done editing the subreddits file, we call index() (the start of program) again. 
+    index("","","")
+
+    
+def addSubreddit(subreddit, name, type):
+    from resources.lib.utils import this_is_a_multihub, format_multihub
+    log( 'addSubreddit ' + subreddit)
+    alreadyIn = False
+    fh = open(subredditsFile, 'r')
+    content = fh.readlines()
+    fh.close()
+    
+    if subreddit:
+        for line in content:
+            #log('line=['+line+']toadd=['+subreddit+']')
+            if line.strip()==subreddit.strip():
+                #log('  MATCH '+line+'='+subreddit)
+                alreadyIn = True
+        if not alreadyIn:
+            fh = open(subredditsFile, 'a')
+            fh.write(subreddit+'\n')
+            fh.close()
+    else:
+        #dialog = xbmcgui.Dialog()
+        #ok = dialog.ok('Add subreddit', 'Add a subreddit (videos)','or  Multiple subreddits (music+listentothis)','or  Multireddit (/user/.../m/video)')
+        #would be great to have some sort of help to show first time user here
+        
+        keyboard = xbmc.Keyboard('', translation(30001))
+        keyboard.doModal()
+        if keyboard.isConfirmed() and keyboard.getText():
+            subreddit = keyboard.getText()
+
+            #cleanup user input. make sure /user/ and /m/ is lowercase
+            if this_is_a_multihub(subreddit):
+                subreddit = format_multihub(subreddit)
+            
+            for line in content:
+                if line.lower()==subreddit.lower()+"\n":
+                    alreadyIn = True
+            if not alreadyIn:
+                fh = open(subredditsFile, 'a')
+                fh.write(subreddit+'\n')
+                fh.close()
+        xbmc.executebuiltin("Container.Refresh")
+
+#MODE removeSubreddit      - name, type not used
+def removeSubreddit(subreddit, name, type):
+    log( 'removeSubreddit ' + subreddit)
+     
+    fh = open(subredditsFile, 'r')
+    content = fh.readlines()
+    fh.close()
+    contentNew = ""
+    for line in content:
+        if line!=subreddit+'\n':
+            #log('line='+line+'toremove='+subreddit)
+            contentNew+=line
+    fh = open(subredditsFile, 'w')
```

Comment:
It's a good practice to use the with statement for file IO http://effbot.org/zone/python-with-statement.htm